python/cudf/cudf/core/dataframe.py

```diff
@@ -665,38 +665,47 @@ class DataFrame(IndexedFrame, Serializable, GetAttrGetItemMixin):
     def __init__(
         self, data=None, index=None, columns=None, dtype=None, nan_as_null=True
     ):
-        super().__init__()
+        col_is_rangeindex = False
```

Comment:
**suggestion/discussion point** Even after this heroic refactoring to make things clearer, this `__init__` method is still _very long_. I haven't yet reviewed everything in detail because I found it quite hard to follow when things are preprocessing to deliver information to a later part of the function compared with preprocessing to produce the final result.

Hence, would it make sense to write the different cases as free functions (or `@staticmethod`s) so that we have something that then looks like:

```
if case_a:
   preprocessed_args = handle_case_a(...)
elif case_b:
   preprocessed_args = handle_case_b(...)

# or whatever
super().__init__(preprocessed_args)
```

WDYT?